include/envoy/event/file_event.h

```diff
@@ -18,15 +18,26 @@ struct FileReadyType {
   static const uint32_t Closed = 0x4;
 };
 
-enum class FileTriggerType { Level, Edge };
+enum class FileTriggerType { Level, Edge, EmulatedEdge };
 
-static constexpr FileTriggerType PlatformDefaultTriggerType
-#ifdef WIN32
-    // Libevent only supports Level trigger on Windows.
-    {FileTriggerType::Level};
+// For POSIX developers to get the Windows behavior of file events
+// you need to add the following definition:
+// `FORCE_LEVEL_EVENTS`
+// You can do this with bazel if you add the following build/test options
+// `--copt="-DFORCE_LEVEL_EVENTS"`
+constexpr FileTriggerType determinePlatformPreferredEventType() {
+#if defined(WIN32) || defined(FORCE_LEVEL_EVENTS)
+#ifdef DO_NOT_OPTIMIZE_LEVEL_EVENTS
```

Comment:
Can this be removed? Why would anyone do this?

Reply:
I removed it. I had lying around to change back and forth while developing but I dont think anyone would want to go back now.